include/options.hxx

```diff
@@ -661,6 +666,13 @@ public:
     return *this;
   }
   
+  friend bool operator==(const Options& lhs, const Options& rhs) {
+    if (lhs.is_value and rhs.is_value) {
```

Comment:
Is it worth having a shortcut here to check if they are the same object `(&lhs == &rhs)`?